sources/osg/Program.js

```diff
@@ -3,12 +3,39 @@ import notify from 'osg/notify';
 import GLObject from 'osg/GLObject';
 import StateAttribute from 'osg/StateAttribute';
 import ShaderProcessor from 'osgShader/ShaderProcessor';
+import Shader from 'osg/Shader';
 import Timer from 'osg/Timer';
 
 // singleton
 var sp = new ShaderProcessor();
 var errorCallback;
 
+var uniformRegexp = /uniform\s+\w+\s+(\w+)((\s)?\[(.*?)\])?/g;
+var attributesRegexp = /(in|attribute)\s+\w+\s+(\w+)\s*;/g;
+
+var getAttributeList = function(shaderText) {
+    var attributeMap = {};
+    var r, attr;
+    while ((r = attributesRegexp.exec(shaderText)) !== null) {
+        attr = r[2];
+        attributeMap[attr] = true;
+    }
+
+    return attributeMap;
+};
+
+var getUniformList = function(shaderText) {
+    var uniformMap = {};
+    var r;
+    while ((r = uniformRegexp.exec(shaderText)) !== null) {
+        var uniform = r[1];
+        uniformMap[uniform] = true;
+    }
+    return uniformMap;
+};
+
+var limitSec = 1;
```

Comment:
not a fan of having this global. Also would be nice to have it configurable.

Reply:
ha yes, all timing code not intended to be in final PR
( debug/bench code )